### DISCUSSION TITLE
Handle `alternative` icon version for svg

### DISCUSSION THREAD

**Discussion by BenSouchet** [devicons/devicon#1040] [Ideas]
Hi,
The title isn't super explicite, so I will try to explain it better.  
Sometimes when looking at brand guidelines to add a new icon to the Devicon project there is an alternative original logo or an alternative plain logo, it would be top that these alternative versions can be handled by the devicon internal process, to be added to the final font.

There is lot of examples, the more recent is the Jekyll logo, there is to Original Wordmark available (light & dark):

## Original Wordmark
<img width="878" alt="01" src="https://user-images.githubusercontent.com/17025808/153176128-6ded26be-60b8-479b-af02-e2645f3517dc.png">

## Original Alternative Wordmark ***(currently not handled by Devicon)***
<img width="876" alt="02" src="https://user-images.githubusercontent.com/17025808/153176501-feaff3d7-5e49-4c43-92fc-6e45adf304ed.png">

## Plain Wordmark
<img width="897" alt="03" src="https://user-images.githubusercontent.com/17025808/153176305-ca5f4145-a9bc-42e7-92d5-a9b239c07577.png">

## Original
<img width="387" alt="04" src="https://user-images.githubusercontent.com/17025808/153176571-c74e0882-77f4-4c41-8740-2149b55655fd.png">

## Plain
<img width="416" alt="05" src="https://user-images.githubusercontent.com/17025808/153176599-caced158-f4d7-483f-aff2-686379c0b51b.png">

Currently possible versions are:
 - `original.svg`
 - `plain.svg`
 - `line.svg`
 - `original-wordmark.svg`
 - `plain-wordmark.svg`
 - `line-wordmark.svg`

The `alternative` keyword would add to possible versions:
 - `original-alternative.svg`
 - `plain-alternative.svg`
 - `line-alternative.svg`
 - `original-alternative-wordmark.svg`
 - `plain-alternative-wordmark.svg`
 - `line-alternative-wordmark.svg`

Tell me that do you think @amacado @Thomas-Boi @Panquesito7 , maybe there is another way to handle these situation when the brand have alternative logo for light & dark theme or when they simply have alternative plain icon.

**Comment by amacado** [comment]
Interesting topic @BenSouchet! I'm not yet convinced that this is a common issue/feature. For further discussion it might be usefull to have some more examples? When we would decide to add those alternatives we should discuss how to name/classify them and in this case we may find some common patterns which we could use to create a name schema?

On the other hand having multiple "alternative" icons could blow up the size and might lead into some organisational issues?

**Reply by BenSouchet** [reply]
First example (of another logo with alternative) is Pytorch logo with two version of the wordmark (light and dark):
https://pytorch.org/assets/images/logo.svg
https://pytorch.org/assets/images/logo-dark.svg

And there is the case when a new logo appear for a new version of tool but lot of people still want the old logo for example the powershell case.
The logo everybody know:
<img width="157" alt="Screenshot 2022-02-09 at 19 48 03" src="https://user-images.githubusercontent.com/17025808/153269329-1d953acf-6577-42cc-bb5b-d1800074ab1e.png">
But this is the current logo (powershell version 7 and above):
<img width="153" alt="Screenshot 2022-02-09 at 19 48 12" src="https://user-images.githubusercontent.com/17025808/153269393-34b3a737-1411-48d8-a775-0e2b988e7726.png">

We can of course duplicate the entry in the json file but for me the true solution is to put the "black" one (since is the actual logo) to original and the blue one as an alternative.

**Comment by Thomas-Boi** [comment]
Having alternatives seem reasonable to me. However, as amacado said, I'm worried about how best to name the alternative icons. A possible idea is using a numbering scheme, so there is `jekyll-original` and `jekyll-original-alt1` and `jekull-original-alt2`. This would accommodate logos that have multiple alternatives.

On an unrelated note @BenSouchet, please let us know your thoughts for the `elasticsearch` PR. We are making a build soon and it'd be great if we can close all our PRs before it.

**Reply by BenSouchet** [reply]
Yes, the more robust and futur proof way is to be able (for example like you said with number iteration) to add multiple alternative icons but if we had only one alternative slot this will be already a great improvement.
